src/library/autodj/autodjprocessor.cpp

```diff
@@ -372,34 +390,34 @@ AutoDJProcessor::AutoDJError AutoDJProcessor::toggleAutoDJ(bool enable) {
         // TODO: This is a total bandaid for making Auto DJ work with decks 3
         // and 4.  We should design a nicer way to handle this.
         for (int i = 2; i < m_decks.length(); ++i) {
-            if (m_decks[i] != NULL && m_decks[i]->isPlaying()) {
+            if (m_decks[i] && m_decks[i]->isPlaying()) {
+                // Keep the current state.
                 return ADJ_DECKS_3_4_PLAYING;
             }
         }
```

Comment:
Could you do some housekeeping and make this a range-based loop like here? https://github.com/mixxxdj/mixxx/pull/4698/files#diff-0468b47d7671b5286de23e574e3b1062d8fe118040f4cab3921ce60265eba4a6R570

Reply:
This loop starts at i = 2.

Reply:
Ah, should've taken a close look at the actual body. my fault.